mixer/mixer.go

```diff
@@ -180,18 +232,19 @@ func (m *Mixer) mixUpdate() {
 		// In case scheduler stops us for too long, we will detect it and run mix multiple times.
 		// This happens if we get scheduled by OS/K8S on a lot of CPUs, but for a very short time.
 		if dt := now.Sub(m.lastMixEndTs); dt > 0 {
+			dt += m.tickerDur / 4 // Account for wake-up jitter
 			n = int(dt / m.tickerDur)
 			m.lastMixEndTs = m.lastMixEndTs.Add(time.Duration(n) * m.tickerDur)
-			if n == 1 {
+			switch n {
+			case 0: // Baseline lastMixEndTs got set later than necessary
+				m.stats.ZeroMixes.Add(1)
```

Comment:
It was intentional that `n == 0` check is after all ifs - this would have caught case where `dt <= 0` for whatever reason. In the new code, it only catches cases when `dt > 0 && n == 0`.